x-pack/legacy/plugins/index_management/public/app/components/template_form/template_form_schemas.tsx

```diff
@@ -68,14 +68,6 @@ export const schemas: Record<string, FormSchema> = {
             ),
           }),
         },
-        {
```

Comment:
As from my comment on the other PR, we should keep this validation for when we create new template.

Reply:
Copying the original suggestion here for reference:

> So here we have multiple options
> 
> - Add a `skipValidation` prop to `UseField` (that would go along with the `enabled: false`
> - Be more granular and being able to provide validation IDs to be skipped
> `validationsToSkip={['startWithDot']}`
> - Extract and export the `name` config from the schema. Then add it on the UseField without the validations
> 
> ```js
> import { schemas, nameConfig } from '../template_form_schemas';
> const { validations, nameConfigWithoutValidations } = nameConfig;
> 
> // And then use that config when editing
> 
> <UseField
>   path="name"
>   componentProps={{
>     idAria: name.idAria,
>     ['data-test-subj']: name.testSubject,
>     euiFieldProps: { disabled: isEditing },
>   }}
>   config={ isEditing ? nameConfigWithoutValidations : nameConfig }
> />
> ```
> 
> I would go with the last one as it is simpler and we don't need to touch the lib for now. It would be a nice enhancement though.

(https://github.com/elastic/kibana/pull/55145#discussion_r368929456)

Reply:
I put the above in `step_logstics.tsx`, but it says `nameConfig` doesn't exist in `../template_form_schemas`

<img width="818" alt="Screen Shot 2020-01-22 at 3 28 48 PM" src="https://user-images.githubusercontent.com/25088052/72943886-18136100-3d2c-11ea-9f6d-981dc8c8dd56.png">

<img width="1050" alt="Screen Shot 2020-01-22 at 3 28 28 PM" src="https://user-images.githubusercontent.com/25088052/72943877-1184e980-3d2c-11ea-8e68-3eebc9adf338.png">

Reply:
So you need to extract the config from the schema. Like this

```js
// template_form_schema.tsx

import {
  FormSchema,
  FIELD_TYPES,
  VALIDATION_TYPES,
  FieldConfig, // Add this import
} from '../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib';

// ...

export const nameConfig: FieldConfig = {
  type: FIELD_TYPES.TEXT,
  label: i18n.translate('xpack.idxMgmt.templateForm.stepLogistics.fieldNameLabel', {
    defaultMessage: 'Name',
  }),
  validations: [
    {
      validator: emptyField(
        i18n.translate('xpack.idxMgmt.templateValidation.templateNameRequiredError', {
          defaultMessage: 'A template name is required.',
        })
      ),
    },
    {
      validator: containsCharsField({
        chars: ' ',
        message: i18n.translate('xpack.idxMgmt.templateValidation.templateNameSpacesError', {
          defaultMessage: 'Spaces are not allowed in a template name.',
        }),
      }),
    },
    {
      validator: startsWithField({
        char: '_',
        message: i18n.translate('xpack.idxMgmt.templateValidation.templateNameUnderscoreError', {
          defaultMessage: 'A template name must not start with an underscore.',
        }),
      }),
    },
    {
      validator: startsWithField({
        char: '.',
        message: i18n.translate('xpack.idxMgmt.templateValidation.templateNamePeriodError', {
          defaultMessage: 'A template name must not start with a period.',
        }),
      }),
    },
    {
      validator: containsCharsField({
        chars: INVALID_TEMPLATE_NAME_CHARS,
        message: ({ charsFound }) =>
          i18n.translate('xpack.idxMgmt.templateValidation.templateNameInvalidaCharacterError', {
            defaultMessage: 'A template name must not contain the character "{invalidChar}"',
            values: { invalidChar: charsFound[0] },
          }),
      }),
    },
    {
      validator: lowerCaseStringField(
        i18n.translate('xpack.idxMgmt.templateValidation.templateNameLowerCaseRequiredError', {
          defaultMessage: 'The template name must be in lowercase.',
        })
      ),
    },
  ],
};

export const schemas: Record<string, FormSchema> = {
  logistics: {
    name: nameConfig, // use it here
    ...
  },
};

```

Reply:
@sebelga thank you for all the detailed steps. I have a question, if `const { validations, nameConfigWithoutValidations } = nameConfig;` then `nameConfigWithoutValidations` would still always validate the system indices correct? We need to remove the period validation 

```
    {
      validator: startsWithField({
        char: '.',
        message: i18n.translate('xpack.idxMgmt.templateValidation.templateNamePeriodError', {
          defaultMessage: 'A template name must not start with a period.',
        }),
      }),
    },
```
from `nameConfig`

Reply:
@sebelga I've got some help from @alisonelizabeth and sorted out my mistakes and confusion. I've uploaded the modified code. Please help check my code again. Thank you!